### PR TITLE
chore: drop npm-run-all, make command npm client agnostic

### PR DESCRIPTION
**Pull request by fi3ework**
## Summary

Ref: https://github.com/web-infra-dev/rspack/pull/7996#issuecomment-2376801593

- Drop `npm-run-all2`, use pnpm run command
  - `npm run test` and `pnpm run test` work fine
  - `npm run test -- xxx` and `pnpm run test -- xxx` works fine
- Remove some commands relay on `npm-run-all2`. In fact, these commands cannot be executed at all because npm-run-all2 is not installed, so I think these commands are obsolete.

with pnpm's `--stream` flag, the log will be easier to be read:

image

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

File: packages/rspack-test-tools/package.json [hunk 1]
```diff
@@ -18,8 +18,8 @@
     "build:viewer": "rspack build",
     "dev:viewer": "rspack serve",
     "dev": "tsc -b -w",
-    "test": "sh -c 'run-s \"test:* -- $*\"' sh",
-    "testu": "sh -c 'run-s \"test:* -- -u $*\"' sh",
+    "test": "pnpm run --stream /^test:.*/",
+    "testu": "pnpm run --stream /^test:.*/ -u",
```
Comment (LingyuCoder) [inline packages/rspack-test-tools/package.json:22]: does this works well on windows too?

Comment (fi3ework) [reply packages/rspack-test-tools/package.json:22]: I'm not sure, since pnpm is platform-agnostic, I assume it could.

Comment (fi3ework) [reply packages/rspack-test-tools/package.json:22]: I think it's already run on Windows? https://github.com/web-infra-dev/rspack/actions/runs/11066114654/job/30747516820?pr=8005

Comment (LingyuCoder) [reply packages/rspack-test-tools/package.json:22]: not sure if `-u` works well, you can try to use `testu` instead in CI

Comment (fi3ework) [reply packages/rspack-test-tools/package.json:22]: `testu` could run on Windows https://github.com/web-infra-dev/rspack/actions/runs/11090959705/job/30814366848?pr=8005#step:9:123.

I manually changed snapshot, and it passed on Windows.

Google Chrome 2024-09-29 16 43 04

**Review by LingyuCoder — APPROVED** [review]

**Pull request merged**